### PR TITLE
Javadoc updates

### PR DESCRIPTION
**Pull request by jweiser**

### CODE REVIEW

File: src/org/gk/database/AttributeTableModel.java [hunk 1]
```diff
@@ -21,7 +21,7 @@ public Class getColumnClass(int col) {
     
     /**
      * Get the SchemaClass object for this AttributeTableModel.
-     * @return
+     * @return SchemaClass object
```
Comment (SolomonShorser-OICR) [inline src/org/gk/database/AttributeTableModel.java]: Could this comment be made slightly more useful? Replace: "object" with "The SchemaClass object for this AttributeTableModel." Yes, I know it's copied-and-pasted from the method description, but the word "object" on its own is not terribly useful, the javadoc might as well not exist (and in this case, I think that could actually be forgiven ;) ).

File: src/org/gk/database/DefaultInstanceEditHelper.java [hunk 2]
```diff
@@ -66,20 +66,20 @@ public Long getDefaultPerson() {
     }
 	
 	/**
-	 * Get the default InstanceEdit. If null, null will be returned.
-	 * @return 
-	 * @see getDefaultInstanceEdit(Window)
+	 * Get the default InstanceEdit. 
+	 * @return Default InstanceEdit. If null, null will be returned.
```
Comment (SolomonShorser-OICR) [inline src/org/gk/database/DefaultInstanceEditHelper.java]: Might be more clear to say "Will return null if this has not yet been set".
(assuming that statement is actually true).

File: src/org/gk/gkCurator/GKCuratorFrame.java [hunk 3]
```diff
@@ -384,9 +384,9 @@ else if (e.getPropertyName().equals("switchType")) {
 	                    GKInstance instance = (GKInstance) e.getNewValue();
 	                    GKSchemaClass oldCls = (GKSchemaClass) e.getOldValue();
 	                    schemaView.switchedType(oldCls, instance);
-	                    eventView.switchedType(oldCls, instance);
-	                    entityView.switchedType(oldCls, instance);
-	                    entityView.switchedType(oldCls, instance);
+	                    eventView.switchedType(instance);
```
Comment (SolomonShorser-OICR) [inline src/org/gk/gkCurator/GKCuratorFrame.java]: This change looks like more than just a javadoc change... what happened here?

File: src/org/gk/gkCurator/SynchronizationDialog.java [hunk 4]
```diff
@@ -191,18 +191,14 @@ private void clearDeleteRecord() {
 		if (deleteList != null) {
 			java.util.List list = deleteList.getSelection();
 			if (list.size() > 0) {
-				try {
```
Comment (SolomonShorser-OICR) [inline src/org/gk/gkCurator/SynchronizationDialog.java]: Why remove the try/catch?

File: src/org/gk/model/ClassAttributeFollowingInstruction.java [hunk 5]
```diff
@@ -14,7 +14,7 @@
  * @author vastrik
  *
  * To change the template for this generated type comment go to
- * Window>Preferences>Java>Code Generation>Code and Comments
+ * Window&gt;Preferences&gt;Java&gt;Code Generation&gt;Code and Comments
```
Comment (SolomonShorser-OICR) [inline src/org/gk/model/ClassAttributeFollowingInstruction.java]: :+1:

File: src/org/gk/persistence/XMLFileAdaptor.java [hunk 6]
```diff
@@ -1076,10 +1100,10 @@ public void removeFromClassMap(GKInstance instance) {
     
     /**
      * Remove the delete record for a list of DB_IDs.
+     * 
      * @param dbIDs a list of DB_IDs.
-     * @throws IOException
      */
-    public void clearDeleteRecord(List<Long> dbIDs) throws IOException {
+    public void clearDeleteRecord(List<Long> dbIDs) {
```
Comment (SolomonShorser-OICR) [inline src/org/gk/persistence/XMLFileAdaptor.java]: Why remove the `throws` declaration?

Comment (SolomonShorser-OICR) [reply src/org/gk/persistence/XMLFileAdaptor.java]: (There are a number of instances of this, but I don't want to comment on each and every one....)

File: src/org/gk/database/AttributeTableModel.java (hunk 1 above)
Comment (jweiser) [reply src/org/gk/database/AttributeTableModel.java]: Fair point - I described the purpose of the SchemaClass object a little bit more in an additional commit.

File: src/org/gk/render/RenderableSwitchHelper.java [hunk 7]
```diff
@@ -110,12 +113,14 @@ private void copyDispayProperties(Renderable source, Renderable target) {
...
-	 * @return
+	 * with the same display name and properties. However, a non-empty reaction cannot switch to a
+	 * pathway - an IllegalArgumentException will be thrown.
+	 * 
+	 * @param node ReactionNode objects to switch to a RenderablePathway object
+	 * @return RenderablePathway object created from the ReactionNode passed
+	 * @throws IllegalArgumentException Thrown if the ReactionNode object contains Node objects
 	 */
-	public RenderablePathway switchReactionToPathway(ReactionNode node) {
+	public RenderablePathway switchReactionToPathway(ReactionNode node) throws IllegalArgumentException {
```
Comment (SolomonShorser-OICR) [inline src/org/gk/render/RenderableSwitchHelper.java]: Why didn't this thrown an exception before? Why add it?

File: src/org/gk/util/BrowserLauncher.java [hunk 8]
```diff
@@ -96,15 +96,7 @@ public static String getBrowser() {
...
-	    File file = null;
-	    try {
-	        file = GKApplicationUtilities.getPropertyFile(PROP_FILE);
-	    }
-	    catch(IOException e) {
-	        System.err.println("BrowserLauncher.getPropertyFile(): " + e);
-	        e.printStackTrace();
-	    }
-	    return file;
+		return GKApplicationUtilities.getPropertyFile(PROP_FILE);
```
Comment (SolomonShorser-OICR) [inline src/org/gk/util/BrowserLauncher.java]: Was the error handling redundant? Is that why this change was made?

File: src/org/gk/database/DefaultInstanceEditHelper.java (hunk 2 above)
Comment (jweiser) [reply src/org/gk/database/DefaultInstanceEditHelper.java]: Rephrased as `@return Default InstanceEdit (or null if none exists).`

File: src/org/gk/gkCurator/GKCuratorFrame.java (hunk 3 above)
Comment (jweiser) [reply src/org/gk/gkCurator/GKCuratorFrame.java]: The method called here and in the two lines below was using `oldCls` as its first argument but on looking at the method body when editing its JavaDocs, I found the parameter was never used.  I removed it from the method so no JavaDocs for the param were needed and so had to change the callers.  I'm going to ask Guanming to review the PR and ask if this branch can be tested running the CuratorTool to make sure everything is okay before I merge.

Comment (SolomonShorser-OICR) [reply src/org/gk/gkCurator/GKCuratorFrame.java]: Ok, that makes sense. I haven't looked at the code in detail so I don't know if it's possible to test this easily with a unit test.

File: src/org/gk/persistence/XMLFileAdaptor.java (hunk 6 above)
Comment (jweiser) [reply src/org/gk/persistence/XMLFileAdaptor.java]: It was removed since there was no place within the method body that throws an IOException.  I checked visually and it was confirmed with Eclipse.

File: src/org/gk/gkCurator/SynchronizationDialog.java (hunk 4 above)
Comment (jweiser) [reply src/org/gk/gkCurator/SynchronizationDialog.java]: The try/catch was removed because the method call wrapped in the try block does not throw the exception being caught.

File: src/org/gk/render/RenderableSwitchHelper.java (hunk 7 above)
Comment (jweiser) [reply src/org/gk/render/RenderableSwitchHelper.java]: It wasn't included in the method header before, but the body does have a statement throwing the exception.  It wasn't forced to be included because it is a runtime/unchecked exception rather than a checked one, but I believe there are other places in the codebase which document IllegalArgumentExceptions, so I included it for consistency.

Comment (SolomonShorser-OICR) [reply src/org/gk/render/RenderableSwitchHelper.java]: Ok.

File: src/org/gk/util/BrowserLauncher.java (hunk 8 above)
Comment (jweiser) [reply src/org/gk/util/BrowserLauncher.java]: Yes, the method called does not throw the exception being caught so I changed its method signature to remove the `throws` portion and removed the error handling here.

Comment (SolomonShorser-OICR) [reply src/org/gk/util/BrowserLauncher.java]: Ok.

**Comment by SolomonShorser-OICR** [comment]
Well, it looks like most of my concerns are resolved. If Guanming can run test the modified code (or if it's possible to test via unit test - though sometimes GUI code can be hard to test that way) and everything passes then I will be OK with this PR.

**Comment by jweiser** [comment]
> I was expecting a PR containing _ONLY_ Javadoc changes, so I'm a bit surprised to see modifications to try/catch blocks, throws declarations. Maybe the IDE flagged them as unncessary/unreachable?
> 
> And then there's this one:
> 
> https://github.com/reactome/CuratorTool/pull/14/files/ac4c8db9d8250beb77a7352ed3c9f159dcc86aa6#r322421040
> 
> What's the background of this change?
> 
> (requesting more detailed explanation, rather than code change, but still not comfortable merging this)

**Comment by jweiser** [comment]
> Well, it looks like most of my concerns are resolved. If Guanming can run test the modified code (or if it's possible to test via unit test - though sometimes GUI code can be hard to test that way) and everything passes then I will be OK with this PR.

Great, thanks!  I'll be sending an e-mail soon to Guanming to ask if he can take a look.

File: src/org/gk/database/util/ChEBIAttributeAutoFiller.java [hunk 9]
```diff
@@ -16,7 +16,7 @@
 import org.gk.persistence.XMLFileAdaptor;
 import org.junit.Test;
 
-import uk.ac.ebi.demo.ols.soap.Query;
+import uk.ac.ebi.ols.soap.Query;
```
Comment (guanmingwu) [inline src/org/gk/database/util/ChEBIAttributeAutoFiller.java]: What version of ols-client are you suggesting? My current version used for the curator tool doesn't have this package.

File: src/org/gk/database/EventCentricViewPane.java [hunk 10]
```diff
@@ -522,10 +522,9 @@ public void rebuildTree() {
 	
 	/**
 	 * Call this method if an Instance's type (i.e. class) is changed.
-	 * @param oldCls the old SchemaClass.
 	 * @param instance the instance whose SchemaClass has been changed.
 	 */
-	public void switchedType(GKSchemaClass oldCls, GKInstance instance) {
+	public void switchedType(GKInstance instance) {
```
Comment (guanmingwu) [inline src/org/gk/database/EventCentricViewPane.java]: Could I keep the original method signature, just for conforming to the same signatures in other views' switchedType(). See line 387 in class GKCuratorFrame.

File: src/org/gk/elv/EntityLevelView.java [hunk 11]
```diff
@@ -470,36 +472,33 @@ public void deleteInstance(GKInstance instance) {
...
-     * @param cls
-     * @param instance
+     * @param instance the instance whose schemaClass has been changed
      */
-    public void switchedType(GKSchemaClass oldCls,
-                             GKInstance instance) {
-        objectListPane.switchedType(oldCls,
-                                    instance);
-        zoomableEditor.switchedType(oldCls, instance);
+    public void switchedType(GKInstance instance) {
```
Comment (guanmingwu) [inline src/org/gk/elv/EntityLevelView.java]: Please see my other comment regarding this method. oldCls may be needed, e.g., to generate an information. Please keep it.

File: src/org/gk/gkCurator/GKCuratorFrame.java [hunk 12]
```diff
@@ -384,9 +384,9 @@ else if (e.getPropertyName().equals("switchType")) {
 	                    GKInstance instance = (GKInstance) e.getNewValue();
 	                    GKSchemaClass oldCls = (GKSchemaClass) e.getOldValue();
 	                    schemaView.switchedType(oldCls, instance);
-	                    eventView.switchedType(oldCls, instance);
-	                    entityView.switchedType(oldCls, instance);
```
Comment (guanmingwu) [inline src/org/gk/gkCurator/GKCuratorFrame.java]: Right there is an extra-call of entityView.switchedType(), which should be deleted. However, please keep one call. Not delete both of these calls!

File: src/org/gk/gkCurator/GKCuratorFrame.java [hunk 13]
```diff
@@ -384,9 +384,9 @@ else if (e.getPropertyName().equals("switchType")) {
 	                    GKInstance instance = (GKInstance) e.getNewValue();
 	                    GKSchemaClass oldCls = (GKSchemaClass) e.getOldValue();
 	                    schemaView.switchedType(oldCls, instance);
-	                    eventView.switchedType(oldCls, instance);
```
Comment (guanmingwu) [inline src/org/gk/gkCurator/GKCuratorFrame.java]: Please see my other comment and keep these lines as they are. Yes, delete one extra line of entityView.switchedType().

File: src/org/gk/persistence/XMLFileAdaptor.java [hunk 14]
```diff
@@ -1122,7 +1122,7 @@ public void clearDeleteRecord(List<Long> dbIDs) {
         // propertyChangeSupport.firePropertyChange("deleteInstance", null, instance);
     }
     
-    public List<?> fetchAllEvents() throws Exception {
+    public List<?> fetchAllEvents() {
```
Comment (guanmingwu) [inline src/org/gk/persistence/XMLFileAdaptor.java]: Is there no exception thrown from this method?

File: src/org/gk/gkCurator/SynchronizationDialog.java (hunk 4 above)
Comment (guanmingwu) [reply src/org/gk/gkCurator/SynchronizationDialog.java]: Have you pushed the changes in XMLFileAdaptor into GitHub?

File: src/org/gk/util/BrowserLauncher.java [hunk 15]
```diff
@@ -96,15 +96,7 @@ public static String getBrowser() {
 	}
 	
 	private static File getPropertyFile() {
-	    File file = null;
```
Comment (guanmingwu) [inline src/org/gk/util/BrowserLauncher.java]: Joel, Actually I'd like to make a change in class GKApplicationUtilities:

	 public static File getPropertyFile(String fileName) throws IOException {
	     File file = new File(REACTOME_DIR + File.separator + fileName);
	     return file;
	 }

and add a check to make this this property exits, which is very important before return:

if (!file.exist()) throw new FileNotFoundException. 

By doing this, we don't need to make other changes either.

File: src/org/gk/gkCurator/SynchronizationDialog.java (hunk 4 above)
Comment (jweiser) [reply src/org/gk/gkCurator/SynchronizationDialog.java]: Yes, it is part of this PR: https://github.com/reactome/CuratorTool/pull/14/commits/c72b0ec0f41a31f33c81045eee5f8f72fd4772d8#diff-fbb43a391912c3c0eb5a8816ac6c5cfcR1106

File: src/org/gk/database/EventCentricViewPane.java (hunk 10 above)
Comment (jweiser) [reply src/org/gk/database/EventCentricViewPane.java]: Sure, if you prefer we can keep the original method signature but I had noticed that the switchedType method here and other calls around line 387 in the GKCuratorFrame class do not use the `oldCls` parameter.  I'm just concerned it is misleading to include it as a parameter.

File: src/org/gk/gkCurator/GKCuratorFrame.java (hunk 13 above)
Comment (jweiser) [reply src/org/gk/gkCurator/GKCuratorFrame.java]: Okay, the extra line has been removed: https://github.com/reactome/CuratorTool/pull/14/commits/d6d55c0af796c78fe445868c70f67d5d91acef4a

File: src/org/gk/persistence/XMLFileAdaptor.java (hunk 14 above)
Comment (jweiser) [reply src/org/gk/persistence/XMLFileAdaptor.java]: No, no exception is thrown.  I checked the sequence of methods called in this method's body starting with `fetchInstancesByClass` on line 1833: 

https://github.com/reactome/CuratorTool/pull/14/commits/c72b0ec0f41a31f33c81045eee5f8f72fd4772d8#diff-fbb43a391912c3c0eb5a8816ac6c5cfcR1833

https://github.com/reactome/CuratorTool/pull/14/commits/c72b0ec0f41a31f33c81045eee5f8f72fd4772d8#diff-fbb43a391912c3c0eb5a8816ac6c5cfcR1794

https://github.com/reactome/CuratorTool/pull/14/commits/666e7d62f22701f0fa81d5f3b70f5e186b42e7fc#diff-75d82b36ec4f308fa4a50505e1d758f8R337

and none throw exceptions.  

IntelliJ also confirmed that an exception was not thrown by the method.

File: src/org/gk/database/util/ChEBIAttributeAutoFiller.java (hunk 9 above)
Comment (jweiser) [reply src/org/gk/database/util/ChEBIAttributeAutoFiller.java]: The version I have found while compiling the dependencies for assembling a POM file is:
```
<dependency>
    <groupId>uk.ac.ebi.ols</groupId>
    <artifactId>ols-client</artifactId>
    <version>1.18</version>
</dependency>
```
available here: https://mvnrepository.com/artifact/uk.ac.ebi.ols/ols-client/1.18

File: src/org/gk/util/BrowserLauncher.java (hunk 15 above)
Comment (jweiser) [reply src/org/gk/util/BrowserLauncher.java]: Okay, I've added in the logic for the FileNotFoundException and reverted the changes to the try/catch block that was removed here: [ef8dfd](https://github.com/reactome/CuratorTool/pull/14/commits/ef8dfd6652fa8dea67b97ded1bfd16c10146e930?w=1)

File: src/org/gk/database/EventCentricViewPane.java (hunk 10 above)
Comment (guanmingwu) [reply src/org/gk/database/EventCentricViewPane.java]: I guess it will not. Thanks!

Comment (jweiser) [reply src/org/gk/database/EventCentricViewPane.java]: Okay, so I should keep the `oldCls` parameter in these methods and their callers?

Comment (jweiser) [reply src/org/gk/database/EventCentricViewPane.java]: Sure -- the reversions to restore `oldCls` have been committed .
